src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1303 @@
+      message: `Push EVSEs from tenant: ${this.tenant.id}`,
+      module: MODULE_NAME, method: 'pushEvseData',
+      detailedMessages: { payload }
+    });
+
+    const publicCert = this.getClientCertificate(ServerAction.OICP_PUSH_EVSES);
+    const privateKey = this.getPrivateKey(ServerAction.OICP_PUSH_EVSES);
+
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized: false, // (NOTE: this will disable client verification)
+      cert: publicCert,
+      key: privateKey,
+      passphrase: ''
+    });
+
+    // Call Hubject
+    await axios.post(fullUrl, payload, {
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      httpsAgent: httpsAgent
+    }).then(function(response) {
+      console.log(response);
+      pushEvseDataResponse = response.data as OICPAcknowledgment;
+    }).catch((error) => {
+      console.log(error.response);
+    });
+    return pushEvseDataResponse;
+  }
+
+  /**
+   * Push EVSE Status
+   */
+  async pushEvseStatus(evseStatuses: OICPEvseStatusRecord[], actionType: OICPActionType): Promise<OICPAcknowledgment> {
+    let pushEvseStatusResponse: OICPAcknowledgment = {} as OICPAcknowledgment;
+    // Check for input parameter
+    if (!evseStatuses) {
+      throw new BackendError({
+        action: ServerAction.OICP_PUSH_EVSE_STATUSES,
+        message: 'Invalid parameters',
+        module: MODULE_NAME, method: 'pushEvseStatus',
+      });
+    }
+    // Get EVSE Status endpoint url
+    // for QA - environment: https://service-qa.hubject.com/api/oicp/evsepush/v21/operators/{operatorID}/status-records
+    // for PROD - environment: https://service.hubject.com/api/oicp/evsepush/v21/operators/{operatorID}/status-records
+    const fullUrl = this.getEndpointUrl('statuses', ServerAction.OICP_PUSH_EVSE_STATUSES);
+
+    // Build payload
+    const operatorEvseStatus: OICPOperatorEvseStatus = {} as OICPOperatorEvseStatus;
+    operatorEvseStatus.OperatorID = this.getOperatorID(ServerAction.OICP_PUSH_EVSE_STATUSES);
+    operatorEvseStatus.OperatorName = this.tenant.name;
+    operatorEvseStatus.EvseStatusRecord = evseStatuses;
+
+    const payload: OICPPushEvseStatusCpoSend = {} as OICPPushEvseStatusCpoSend;
+    payload.ActionType = actionType;
+    payload.OperatorEvseStatus = operatorEvseStatus;
+
+    // Log
+    Logging.logDebug({
+      tenantID: this.tenant.id,
+      action: ServerAction.OICP_PUSH_EVSE_STATUSES,
+      message: `Push EVSE statuses from tenant: ${this.tenant.id}`,
+      module: MODULE_NAME, method: 'pushEvseStatus',
+      detailedMessages: { payload }
+    });
+
+    const httpsAgent = new https.Agent({
+      rejectUnauthorized: false, // (NOTE: this will disable client verification)
+      cert: this.getClientCertificate(ServerAction.OICP_PUSH_EVSE_STATUSES),
+      key: this.getPrivateKey(ServerAction.OICP_PUSH_EVSE_STATUSES),
+      passphrase: ''
+    });
+
+    // Call Hubject
+    await axios.post(fullUrl, String(JSON.stringify(payload)), {
```

Comment:
Please use axios factory.

Reply:
I tried to use the this.axiosInstance, but always got an error message that I could not fix.
The error says:
"Converting circular structure to JSON
    --> starting at object with constructor 'TLSSocket'
    |     property '_httpMessage' -> object with constructor 'ClientRequest'
    --- property 'socket' closes the circle"

Reply:
I figured out that this error comes from the Response interceptor added to the axiosInstance in the getAxiosInstance method. My Request work, if I comment that out.
// Add a Response interceptor
      axiosInstance.interceptors.response.use((response: AxiosResponse) => {
        Logging.logAxiosResponse(tenantID, response);
        return response;
      }, async (error: AxiosError) => {
        Logging.logAxiosError(tenantID, error);
        return Promise.reject(error);
      });
That should be checked

Reply:
Comment out the code in AxiosFactory in that PR for now with a big fat FIXME. 
We will address that issue later on in AxiosFactory.

Reply:
okay